src/c/types/array.c

```diff
@@ -16,123 +16,176 @@
 
 #include <string.h>
 
-static void ucdr_array_to_buffer(ucdrBuffer* ub, const uint8_t* array, size_t size, size_t data_size);
-static void ucdr_buffer_to_array(ucdrBuffer* ub, uint8_t* array, size_t size, size_t data_size);
+static void ucdr_array_to_buffer(
+        ucdrStream* us,
+        const uint8_t* array,
+        size_t size);
+
+static void ucdr_buffer_to_array(
+        ucdrStream* us,
```

Comment:
It seems like it should be const

Reply:
I cannot be const because `offset`, `iterator` and `buffer_info` could be modified.